src/Manager.php

```diff
@@ -96,122 +96,124 @@ public function userHasPermission($userId, string $permissionName, array $parame
+        $this->itemsStorage->addChild($parentName, $childName);
     }
 
     /**
      * Removes a child from its parent.
      * Note, the child item is not deleted. Only the parent-child relationship is removed.
      *
-     * @param Item $parent The parent item.
-     * @param Item $child The child item.
+     * @param string $parentName The name of the parent item.
+     * @param string $childName The name of the child item.
      */
-    public function removeChild(Item $parent, Item $child): void
+    public function removeChild(string $parentName, string $childName): void
     {
-        if ($this->hasChild($parent, $child)) {
-            $this->rolesStorage->removeChild($parent, $child);
+        if ($this->hasChild($parentName, $childName)) {
+            $this->itemsStorage->removeChild($parentName, $childName);
         }
     }
 
     /**
      * Removes all children form their parent.
      * Note, the children items are not deleted. Only the parent-child relationships are removed.
      *
-     * @param Item $parent The parent item.
+     * @param string $parentName The name of the parent item.
      */
-    public function removeChildren(Item $parent): void
+    public function removeChildren(string $parentName): void
     {
-        if ($this->rolesStorage->hasChildren($parent->getName())) {
-            $this->rolesStorage->removeChildren($parent);
+        if ($this->itemsStorage->hasChildren($parentName)) {
+            $this->itemsStorage->removeChildren($parentName);
         }
     }
 
     /**
      * Returns a value indicating whether the child already exists for the parent.
      *
-     * @param Item $parent The parent item.
-     * @param Item $child The child item.
+     * @param string $parentName The name of the parent item.
+     * @param string $childName The name of the child item.
      *
      * @return bool Whether `$child` is already a child of `$parent`
      */
-    public function hasChild(Item $parent, Item $child): bool
+    public function hasChild(string $parentName, string $childName): bool
     {
-        return array_key_exists($child->getName(), $this->rolesStorage->getChildrenByName($parent->getName()));
+        return array_key_exists($childName, $this->itemsStorage->getChildren($parentName));
     }
 
     /**
      * Assigns a role or permission to a user.
      *
-     * @param Item $item Role or permission to be assigned.
+     * @param string $itemName Name of the role or the permission to be assigned.
      * @param string $userId The user ID.
      *
-     * @throws Exception If the role or permission has already been assigned to the user.
+     *@throws Exception If the role or permission has already been assigned to the user.
      *
      * @return Assignment|null The role or permission assignment information.
      */
-    public function assign(Item $item, string $userId): ?Assignment
+    public function assign(string $itemName, string $userId): ?Assignment
```

Comment:
We should user same order both in items repository and in manager. May be change order in `AssignmentsStorageInterface::add()` and `AssignmentsStorageInterface::get()` ?

Reply:
Adjusted `add()` and `get()`.